src/nonePublicStuff.ts

```diff
@@ -0,0 +1,345 @@
+        })
+      )
+    );
+  } else if (requestIntermediate.type === DELETE_MANY) {
+    rawResponse = await Promise.all(
+      requestIntermediate.params.ids.map((id: any) =>
+        httpClient(`${apiUrl}/${parsedResource.realResource}/${id}`, {
+          method: "DELETE"
+        })
+      )
+    );
+  } else {
+    const { url, options } = convertDataRequestToHTTP(
+      apiUrl,
+      requestIntermediate.type,
+      // Original and not intermediate here.
+      resource,
+      requestIntermediate.params
+    );
+
+    rawResponse = await httpClient(url, options);
+  }
+
+  // maybe pass here original and not intermediate values?
+  let responseIntermediate = {
+    response: rawResponse,
+    type: requestIntermediate.type,
+    resource: requestIntermediate.resource,
+    params: requestIntermediate.params
+  };
+
+  if (configuration && configuration.responseMutator) {
+    responseIntermediate = configuration.responseMutator(
+      responseIntermediate,
+      requestIntermediate
+    );
+  }
+
+  const oneMoreIntermediateResponse = convertHTTPResponse(
+    responseIntermediate.response,
+    requestIntermediate
+  );
+
+  // in some conditions, we want to fetch fresh copy of the data from nest/crud.
+  switch (requestIntermediate.type) {
+    case "CREATE":
+    case "UPDATE":
+      return makeNestjsxCrudRequest(
+        apiUrl,
+        httpClient,
+        configuration,
+        "GET_ONE",
+        resource,
+        { id: oneMoreIntermediateResponse.data.id }
+      );
+
+    case "UPDATE_MANY":
+      return Promise.all(
+        oneMoreIntermediateResponse.data.map(({ id }: { id: any }) =>
+          makeNestjsxCrudRequest(
+            apiUrl,
+            httpClient,
+            configuration,
+            "GET_ONE",
+            resource,
+            { id }
+          )
+        )
+      ).then((all: any) => ({
+        data: all
+      }));
+
+    default:
+      return oneMoreIntermediateResponse;
+  }
+}
+
+/** @internal */
+export function convertHTTPResponse(
+  rawResponse: any,
+  request: ClientRequestType
+) {
+  let toReturn: any = Array.isArray(rawResponse)
+    ? { data: rawResponse.map(entry => entry.json) }
+    : { data: rawResponse.json };
+
+  switch (request.type) {
+    case DELETE_MANY:
+      toReturn = {
+        data: request.params.ids
+      };
+      break;
+
+    case GET_LIST:
+    case GET_MANY_REFERENCE:
+      toReturn = {
+        data: rawResponse.json.data,
+        total: rawResponse.json.total
+      };
+  }
+
+  return toReturn;
+}
+
+/** @internal */
+export function composeFilter(paramsFilter: any) {
+  const flatFilter = fetchUtils.flattenObject(paramsFilter);
+  const filter = Object.keys(flatFilter).map(key => {
+    const splitKey = key.split("||");
+    const ops = splitKey[1] ? splitKey[1] : CondOperator.CONTAINS;
+    let field = splitKey[0];
+
+    if (field.indexOf("_") === 0 && field.indexOf(".") > -1) {
```

Comment:
Please add some comments for such fancy code blocks.
`indexOf() !== -1` is better option here.